include/flamegpu/simulation/CUDAEnsemble.h

```diff
@@ -22,6 +23,20 @@ struct RunLog;
  */
 class CUDAEnsemble {
  public:
+#ifdef FLAMEGPU_ENABLE_MPI
```

Comment:
I don't really understand what you mean.

The ifdef isn't strictly required, there's little harm (other than confusion) for the below `EnvelopeTag` enum being available everywhere.

Reply:
The comment was meant to be on the enum line I expect.

`enum class` enums are type safe - i.e. you can't set it to `0`/`1` / some other enum that happens to map to `0`/`1`, you have to provide it an `EnvelopeTag`, or cast the value. 

They are just nicer IMO. 

I think I was probably asking if there was a reason e.g. do they get fed into MPI which expects the underlying int, which would require the casting from the enum class to int or not (given that MPI is a C library not c++ (since the c++ interface was removed in/after 2.0? as it caused more problems than it sovled)

Reply:
Don't think im aware of `enum class`.

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum3-prefer-class-enums-over-plain-enums